### ISSUE TITLE
Include jupyterlab-s3-browser extension

### ISSUE THREAD

**Issue by MichaelClifford** [thoth-station/s2i-minimal-notebook#54]
**Is your feature request related to a problem? Please describe.**

I'm always frustrated when I have to use the s3cmd command line tool to see what's in my remote storage. It would be great if I could navigate ceph the same way I navigate my PVC files.   

**Describe the solution you'd like**

I would like the [jupyterlab-s3-browser extension](https://github.com/IBM/jupyterlab-s3-browser) added to the jupyterlab notebooks. 

**Describe alternatives you've considered**

No other alternatives have been proposed for GUI lab/notebook s3 exploration. There are CLI tools, python API's and the Hue S3 browser. However, I think this lab extension is far faster, easier and more reliable to use than all of these alternatives.  

**Additional context**

**Comment by pacospace** [comment]
> **Is your feature request related to a problem? Please describe.**
> 
> I'm always frustrated when I have to use the s3cmd command line tool to see what's in my remote storage. It would be great if I could navigate ceph the same way I navigate my PVC files.
> 
> **Describe the solution you'd like**
> 
> I would like the [jupyterlab-s3-browser extension](https://github.com/IBM/jupyterlab-s3-browser) added to the jupyterlab notebooks.
> 
> **Describe alternatives you've considered**
> 
> No other alternatives have been proposed for GUI lab/notebook s3 exploration. There are CLI tools, python API's and the Hue S3 browser. However, I think this lab extension is far faster, easier and more reliable to use than all of these alternatives.
> 
> **Additional context**
> Add any other context or screenshots about the feature request here.

Thanks @MichaelClifford!! Really nice extension, I think it can be interesting also for ODH downstream!  I don't see big constraints to have it.

**Comment by pacospace** [comment]
cc @crobby @guimou @cfchase

**Comment by LaVLaS** [comment]
I think we should investigate adding this to [s2i-generic-data-science-notebook](https://github.com/thoth-station/s2i-generic-data-science-notebook) since it is including the boto3 package with the expectation that a user will need an s3 library

**Comment by pacospace** [comment]
> I think we should investigate adding this to [s2i-generic-data-science-notebook](https://github.com/thoth-station/s2i-generic-data-science-notebook) since it is including the boto3 package with the expectation that a user will need an s3 library

It will be there if we add it to s2i-minimal-notebook as it is the base image for that, while @MichaelClifford requested it for the s2i-lab-elyra image.

**Comment by cfchase** [comment]
> I think we should investigate adding this to [s2i-generic-data-science-notebook](https://github.com/thoth-station/s2i-generic-data-science-notebook) since it is including the boto3 package with the expectation that a user will need an s3 library

Good point.  I'm wondering if adding the s3 extension makes this minimal notebook more than just minimal?  Adding it to the s2i-generic-data-science-notebook would also add it to elyra, would it not?

**Comment by pacospace** [comment]
> > I think we should investigate adding this to [s2i-generic-data-science-notebook](https://github.com/thoth-station/s2i-generic-data-science-notebook) since it is including the boto3 package with the expectation that a user will need an s3 library
> 
> Good point. I'm wondering if adding the s3 extension makes this minimal notebook more than just minimal? Adding it to the s2i-generic-data-science-notebook would also add it to elyra, would it not?

Nope, they use same base image (quay.io/thoth-station/s2i-minimal-py38-notebook) but they have different packages.

**Comment by pacospace** [comment]
> > I think we should investigate adding this to [s2i-generic-data-science-notebook](https://github.com/thoth-station/s2i-generic-data-science-notebook) since it is including the boto3 package with the expectation that a user will need an s3 library
> 
> Good point. I'm wondering if adding the s3 extension makes this minimal notebook more than just minimal? Adding it to the s2i-generic-data-science-notebook would also add it to elyra, would it not?

Good point for not being minimal anymore, but I guess boto3 should be a minimal package to interact with s3 compatible object storages. Or we can create a layer on top of this for jupyterlab extensions required by users.

**Comment by harshad16** [comment]
very good points all
 - The Elyra image is using minimal as base 
 -  The generic would be base for Tensorflow and PyTorch
 
Maybe it would be good to make generic base for elyra as well , then user of the elyra would have other ds related packages to use.

**Comment by harshad16** [comment]
great i see some +1 on the previous comment
i will move this to the generic notebook and get started there.
